config/opal_setup_cc.m4

```diff
@@ -243,47 +245,34 @@ AC_DEFUN([OPAL_SETUP_CC],[
 
     # Do we want code coverage
     if test "$WANT_COVERAGE" = "1"; then
-        if test "$opal_c_vendor" = "gnu" ; then
-            # For compilers > gcc-4.x, use --coverage for
-            # compiling and linking to circumvent trouble with
-            # libgcov.
-            CFLAGS_orig="$CFLAGS"
-            LDFLAGS_orig="$LDFLAGS"
-
-            CFLAGS="$CFLAGS_orig --coverage"
-            LDFLAGS="$LDFLAGS_orig --coverage"
-            OPAL_COVERAGE_FLAGS=
-
-            AC_CACHE_CHECK([if $CC supports --coverage],
-                      [opal_cv_cc_coverage],
-                      [AC_TRY_COMPILE([], [],
-                                      [opal_cv_cc_coverage="yes"],
-                                      [opal_cv_cc_coverage="no"])])
-
-            if test "$opal_cv_cc_coverage" = "yes" ; then
-                OPAL_COVERAGE_FLAGS="--coverage"
-                CLEANFILES="*.gcno ${CLEANFILES}"
-                CONFIG_CLEAN_FILES="*.gcda *.gcov ${CONFIG_CLEAN_FILES}"
-            else
-                OPAL_COVERAGE_FLAGS="-ftest-coverage -fprofile-arcs"
-                CLEANFILES="*.bb *.bbg ${CLEANFILES}"
-                CONFIG_CLEAN_FILES="*.da *.*.gcov ${CONFIG_CLEAN_FILES}"
-            fi
-            CFLAGS="$CFLAGS_orig $OPAL_COVERAGE_FLAGS"
-            LDFLAGS="$LDFLAGS_orig $OPAL_COVERAGE_FLAGS"
-            OPAL_WRAPPER_FLAGS_ADD([CFLAGS], [$OPAL_COVERAGE_FLAGS])
-            OPAL_WRAPPER_FLAGS_ADD([LDFLAGS], [$OPAL_COVERAGE_FLAGS])
-
-            OPAL_FLAGS_UNIQ(CFLAGS)
-            OPAL_FLAGS_UNIQ(LDFLAGS)
+        # For compilers > gcc-4.x, use --coverage for
+        # compiling and linking to circumvent trouble with
+        # libgcov.
+        LDFLAGS_orig="$LDFLAGS"
+        LDFLAGS="$LDFLAGS_orig --coverage"
+        OPAL_COVERAGE_FLAGS=
+
+        _OPAL_CHECK_SPECIFIC_CFLAGS(--coverage, coverage)
+        if test "$opal_cv_cc_coverage" = "1" ; then
+            OPAL_COVERAGE_FLAGS="--coverage"
+            CLEANFILES="*.gcno ${CLEANFILES}"
+            CONFIG_CLEAN_FILES="*.gcda *.gcov ${CONFIG_CLEAN_FILES}"
             AC_MSG_WARN([$OPAL_COVERAGE_FLAGS has been added to CFLAGS (--enable-coverage)])
-
-            WANT_DEBUG=1
         else
-            AC_MSG_WARN([Code coverage functionality is currently available only with GCC])
-            AC_MSG_ERROR([Configure: Cannot continue])
-       fi
-    fi
+            _OPAL_CHECK_SPECIFIC_CFLAGS(-ftest-coverage, ftest_coverage)
+            _OPAL_CHECK_SPECIFIC_CFLAGS(-fprofile-arcs, fprofile_arcs)
+            if test "$opal_cv_cc_ftest_coverage" = "0" || test "opal_cv_cc_fprofile_arcs" = "0" ; then
+                AC_MSG_WARN([Code coverage functionality is not currently available with $CC])
+                AC_MSG_ERROR([Configure: Cannot continue])
+            fi
+            CLEANFILES="*.bb *.bbg ${CLEANFILES}"
+            OPAL_COVERAGE_FLAGS="-ftest-coverage -fprofile-arcs"
+        fi
```

Comment:
Don't you want to set LDFLAGS back to LDFLAGS_orig in an else case if the compiler doesn't support `--coverage` flag for some reason?

Reply:
If the user requested coverage and no --coverage is supported, the previous logic would throw an error and abort. I just kept that logic, so no need to reset anything.